streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -598,6 +605,15 @@ public void checkpoint() {
         // checkpoint those stores that are only logged and persistent to the checkpoint file
         final Map<TopicPartition, Long> checkpointingOffsets = new HashMap<>();
         for (final StateStoreMetadata storeMetadata : stores.values()) {
+            if (storeMetadata.checkpointCallback != null && !storeMetadata.corrupted) {
+                try {
+                    storeMetadata.checkpointCallback.checkpoint();
+                } catch (final IOException e) {
+                    throw new ProcessorStateException(format("%sError creating position checkpoint file",
```

Comment:
```suggestion
                    throw new ProcessorStateException(format("%sError checkpointing store",
```
Also, it might be nice to include the store name and partition in the error message.